src/chem/modis_aod2ioda.py

```diff
@@ -0,0 +1,183 @@
+
+            # convert time to date_string
+            lats = ncd.variables['Latitude'][:].ravel()
+            lons = ncd.variables['Longitude'][:].ravel()
+            aod = ncd.variables['AOD_550_Dark_Target_Deep_Blue_Combined'][:].ravel()
+            land_sea_flag = ncd.variables['Land_sea_Flag'][:].ravel()
+            QC_flag = ncd.variables['Land_Ocean_Quality_Flag'][:].ravel()
+            sol_zen = ncd.variables['Solar_Zenith'][:].ravel()
+            sol_zen = sol_zen
+            sen_zen = ncd.variables['Sensor_Zenith'][:].ravel()
+            sen_zen = sen_zen
+            unc_land = ncd.variables['Deep_Blue_Aerosol_Optical_Depth_550_Land_Estimated_Uncertainty'][:].ravel()
+            unc_land = unc_land
+
+            time_1 = timedelta(seconds=1)
+            obs_time = np.empty_like(QC_flag, dtype=object)
+            obs_time = datetime.fromisoformat('1993-01-01') + modis_time*time_1
+
+            # uncertainty estimates:
+            # From MODIS file (over ocean) and Levy, 2010 (over land)
+            # flag = 0 (ocean) 1(land) 2(coastal)
+
+            over_ocean = np.logical_not(land_sea_flag > 0)
+            over_land = np.logical_not(land_sea_flag == 0)
+            UNC = np.where(over_land, unc_land, np.add(0.05, np.multiply(0.15, aod)))
+            UNC_2 = UNC[0]
+
+            if first:
+                self.loc_mdata['latitude'] = lats
+                self.loc_mdata['longitude'] = lons
+                self.loc_mdata['datetime'] = self.writer.FillNcVector(obs_time, "datetime")
+                self.loc_mdata['latitude'] = np.concatenate((self.loc_mdata['latitude'], lats[0:1]))
+                self.loc_mdata['longitude'] = np.concatenate((self.loc_mdata['longitude'], lons[0:1]))
+                self.loc_mdata['datetime'] = np.concatenate((self.loc_mdata['datetime'],
+                                                            self.writer.FillNcVector(obs_time[0:1], "datetime")))
+            else:
+                self.loc_mdata['latitude'] = np.concatenate((self.loc_mdata['latitude'], lats))
+                self.loc_mdata['longitude'] = np.concatenate((self.loc_mdata['longitude'], lons))
+                self.loc_mdata['datetime'] = np.concatenate((self.loc_mdata['datetime'],
+                                                            self.writer.FillNcVector(obs_time, "datetime")))
+
+            for ncvar, iodavar in obsvars.items():
+                data = aod.astype('float32')
+                err = UNC
+                err_2 = err[0]
+                if first:
+                    self.outdata[self.varDict[iodavar]['valKey']] = aod
+                    self.outdata[self.varDict[iodavar]['errKey']] = err
+                    self.outdata[self.varDict[iodavar]['qcKey']] = QC_flag
+                    self.outdata[self.varDict[iodavar]['valKey']] = np.concatenate(
+                        (self.outdata[self.varDict[iodavar]['valKey']], data[0:1]))
```

Comment:
Same question as above, why the extra point concatenation? I know you mentioned it in the PR description but it seems like it is not correct.

Reply:
See the above discussion.